modules/Settings/Picklist/models/Field.php

```diff
@@ -129,4 +129,95 @@ public function validate($value, $id = false)
 			throw new \App\Exceptions\AppException(\App\Language::translateArgs('ERR_DUPLICATES_VALUES_FOUND', 'Other.Exceptions', $value), 513);
 		}
 	}
+
+	/**
+	 * Is process status field.
+	 *
+	 * @return bool
+	 */
+	public function isProcessStatusField(): bool
+	{
+		return $this->getFieldParams()['isProcessStatusField'] ?? false;
+	}
+
+	/**
+	 * Update record status.
+	 *
+	 * @param int   $id
+	 * @param int   $recordState
+	 * @param int[] $timeCounting
+	 *
+	 * @throws \App\Exceptions\AppException
+	 *
+	 * @return bool
+	 */
+	public function updateRecordStatus(int $id, int $recordState, array $timeCounting): bool
+	{
+		if (!$this->isProcessStatusField()) {
+			throw new \App\Exceptions\AppException(\App\Language::translate('LBL_IS_NOT_A_PROCESS_STATUS_FIELD', 'Settings:Picklist'), 406);
+		}
+		if (!$this->isEditable()) {
+			throw new \App\Exceptions\AppException(\App\Language::translate('LBL_NON_EDITABLE_PICKLIST_VALUE', 'Settings:Picklist'), 406);
+		}
+		$pickListFieldName = $this->getName();
+		$primaryKey = \App\Fields\Picklist::getPickListId($pickListFieldName);
+		$tableName = \App\Fields\Picklist::getPickListTableName($pickListFieldName);
+		$tabId = $this->get('tabid');
+		$moduleName = \App\Module::getModuleName($tabId);
+		foreach ($timeCounting as $time) {
+			if (!is_int($time)) {
+				throw new \App\Exceptions\IllegalValue('ERR_NOT_ALLOWED_VALUE||' . $time, 406);
+			}
+		}
+		$newTimeCountingValue = \App\RecordStatus::getTimeCountingStringValueFromArray($timeCounting);
+		if ($newTimeCountingValue === ',,') {
```

Comment:
?

Reply:
empty value from `','.implode().','` the easiest way